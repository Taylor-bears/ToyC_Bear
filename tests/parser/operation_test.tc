int main() {
  return 1 + 2 * 3 == 7 && 10 / 5 > 1 || !0;
  // 预期解析顺序:
  // 1. !0
  // 2. 2 * 3 -> 6
  // 3. 10 / 5 -> 2
  // 4. 1 + 6 -> 7
  // 5. 7 == 7 -> true
  // 6. 2 > 1 -> true
  // 7. true && true -> true
  // 8. true || true -> true
}