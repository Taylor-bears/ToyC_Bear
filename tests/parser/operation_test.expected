[FuncDef(name=main, params=[], body=Block([Return(Some(BinOp(Or, BinOp(And, BinOp(Eq, BinOp(Add, Int(1), BinOp(Mul, Int(2), Int(3))), Int(7)), BinOp(Gt, BinOp(Div, Int(10), Int(5)), Int(1))), UnOp(Not, Int(0)))))]))]